dvc/repo/experiments/__init__.py

```diff
@@ -877,6 +877,8 @@ def get_exact_name(self, rev: str):
         Prefers tags, branches (heads), experiments in that orer.
         """
         exclude = f"{EXEC_NAMESPACE}/*"
+        if rev in self.stash_revs:
+            return self.stash_revs[rev].name
```

Comment:
This stash check should be done last - experiments (including stashed experiments) should have lower priority than tags or branch names

Reply:
Reorder here would make the code a little corrupt. And shouldn't the stash ones be mutually exclusive with the commit ones? But I agreed that making commit ones ahead of stash ones is a logical/common choice, I also hesitated here before I wrote them down.

Reply:
It's unlikely that a git tag or branch would ever point to one of our stash commits, but it's technically a legal state - you can manually tag any git commit, and manually create a `refs/heads/branch_name` entry that points to any git commit.
This includes our stash commits, so we have to account for it.